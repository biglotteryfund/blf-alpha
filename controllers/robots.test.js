/* eslint-env jest */
'use strict';

const httpMocks = require('node-mocks-http');
const robotsHandler = require('./robots');

function createRequestFor(host) {
    return httpMocks.createRequest({
        method: 'GET',
        url: '/robots.txt',
        headers: { 'Host': host, 'X-Forwarded-Proto': 'https' },
    });
}

test('should allow indexing on live domain', () => {
    const req = createRequestFor('www.tnlcommunityfund.org.uk');
    const res = httpMocks.createResponse();
    robotsHandler(req, res);
    expect(res._getData()).toMatchInlineSnapshot(`
        "user-agent: *
        sitemap: https://www.tnlcommunityfund.org.uk/sitemap.xml
        disallow: /about/newsletter/insights"
    `);
});

test('should block indexing on other domains', () => {
    const req = createRequestFor('apply.tnlcommunityfund.org.uk');
    const res = httpMocks.createResponse();
    robotsHandler(req, res);
    expect(res._getData()).toMatchInlineSnapshot(`
        "user-agent: *
        sitemap: https://apply.tnlcommunityfund.org.uk/sitemap.xml
        disallow: /about/newsletter/insights
        disallow: /"
    `);
});
