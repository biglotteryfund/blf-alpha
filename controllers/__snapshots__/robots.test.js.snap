// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should allow indexing on live domain 1`] = `
"user-agent: *
sitemap: https://www.tnlcommunityfund.org.uk/sitemap.xml"
`;

exports[`should block indexing on other domains 1`] = `
"user-agent: *
sitemap: https://apply.tnlcommunityfund.org.uk/sitemap.xml
disallow: /"
`;
